### PR TITLE
Improve Add and Edit Allocations date selection

### PR DESCRIPTION
**Pull request by Ovifer13**
**Issue #481**
Sometimes selecting dates from a calendar can be difficult, so we are improving the way people can select weeks, months, and years for when they create or edit an allocation.

### CODE REVIEW

**Comment by Ovifer13** [comment]
This functionality should add weeks, months, and years from the start date to the end date or add the same amount of time to both?
For example. The user sets a starting date and can add weeks, months, and years only to the end date with the new functionality?
image

@otech47 @sofiarm21 @mecoccaro

**Comment by Ovifer13** [comment]
I already did the accordion but Instead of Quick Select what name can we use there?
image

**Comment by Ovifer13** [comment]
**Update**
<a href="https://www.loom.com/share/e7a331c96a1145b9b353a2d5a2b347b0">
    <p>Trinary: Date Quick Select explanation - Watch Video</p>
    
  </a>

File: src/components/DateInput.js [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+                                            variant={`${selectedWeek == 4 
+                                                ? 'contained' 
+                                                : 'outlined'
+                                            }`}
+                                            onClick={() => getWeekMonthYears(4)}
+                                        >
+                                            4
+                                        </Button>
+                                    </Grid>
+                                    <Grid item xs={2} />
```
Comment (sofiarm21) [inline src/components/DateInput.js:128]: To make a more maintainable code do the following
```
const renderWeeks = () => {
   const weeks = [1,2,3,4]
   return weeks.map(w => {
      <Grid item xs={2}>
            <Button 
                    color='primary'
                     variant={`${selectedWeek == w 
                          ? 'contained' 
                           : 'outlined'
                        }`}
                         onClick={() => getWeekMonthYears(w)}
             >
                           {w}
             </Button>
           </Grid>
        <Grid item xs={2} />

   }
}
```

Now all you have to do is call the function `renderWeeks`, this way you are saving a lot of lines of code and if we want to change the weeks for any reason we only have to change the array. Do the same for months and years.

File: src/styles/DateInput.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
+.DateInput {
+    .MuiPaper-elevation1 {
+        box-shadow: none;
+        border: 1px solid #F1F1F1;
+        color: #828282;
```
Comment (sofiarm21) [inline src/styles/DateInput.scss:5]: Use the color variables. See `ContributorTile.scss` for reference

**Review by sofiarm21 — CHANGES_REQUESTED** [review]

File: src/components/DateInput.js (hunk 1 above)
Comment (Ovifer13) [reply src/components/DateInput.js]: Thanks Sofia for this method, the code now looks cleaner!

**Review by sofiarm21 — APPROVED** [review]

**Pull request merged**